src/collection.js

```diff
@@ -365,10 +377,32 @@ export default class Collection {
         "max_target.data.last_modified": String(at),
       },
     });
+    return changes;
+  }
 
+  /**
+   * @private
+   */
+  async getSnapshot(at) {
+    if (!Number.isInteger(at) || at <= 0) {
+      throw new Error("Invalid argument, expected a positive integer.");
+    }
+    // If history doesn't have enough data, reject with an error
+    const oldestHistory = await this.findHistoryOldestTimestamp();
```

Comment:
In my understanding we'd rather need `changes[changes.length + 1]` which is impossible to get. Ok I'm officially burning out with this, wanna pair on it?